components/edit-collective/sections/virtual-cards/filters/MerchantFilter.js

```diff
@@ -0,0 +1,41 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { useRouter } from 'next/router';
+
+import { CollectiveType } from '../../../../../lib/constants/collectives';
+
+import { CUSTOM_OPTIONS_POSITION } from '../../../../CollectivePicker';
+import CollectivePickerAsync from '../../../../CollectivePickerAsync';
+
+function MerchantFilter({ inputId, onChange, virtualCardMerchants }) {
+  const router = useRouter();
+
+  return (
+    <CollectivePickerAsync
+      id={inputId}
+      inputId={inputId}
+      types={[CollectiveType.COLLECTIVE]}
+      maxMenuHeight={200}
+      placeholder="Search for Merchants"
+      preload={true}
+      limit={10}
+      onChange={({ value }) => onChange(value.slug)}
+      value={
+        router.query.merchant
+          ? { label: router.query.merchant, value: router.query.merchant }
+          : { label: 'All Merchants', value: 'ALL' }
+      }
+      emptyCustomOptions={[{ label: 'All Merchants', value: 'ALL' }]}
+      customOptionsPosition={CUSTOM_OPTIONS_POSITION.TOP}
+      collectives={virtualCardMerchants}
```

Comment:
If `collectives` are pre-fetched, than we probably want to use `CollectivePicker` instead of `CollectivePickerAsync`. The goal of `CollectivePickerAsync` is to fetch collectives asynchronously.

Reply:
Thanks done. 👍🏽